test/e2e/pensions-helpers.js

```diff
@@ -170,6 +170,98 @@ function completeDependentAddressInfo(client, data, index) {
+      .click('.va-growable-expanded .float-left');
+}
+
+function completeMonthlyIncomeInfo(client, data) {
+  client
+      .fill('input[name="root_monthlyIncome_socialSecurity"]', data.monthlyIncome.socialSecurity)
+      .fill('input[name="root_monthlyIncome_civilService"]', data.monthlyIncome.civilService)
+      .fill('input[name="root_monthlyIncome_railroad"]', data.monthlyIncome.railroad)
+      .fill('input[name="root_monthlyIncome_blackLung"]', data.monthlyIncome.blackLung)
+      .fill('input[name="root_monthlyIncome_serviceRetirement"]', data.monthlyIncome.serviceRetirement)
+      .fill('input[name="root_monthlyIncome_ssi"]', data.monthlyIncome.ssi)
+      .click('.pensions-sources-add-btn')
+      .fill('input[name="root_monthlyIncome_additionalSources_0_name"]', data.netWorth.additionalSources[0].name)
+      .fill('input[name="root_monthlyIncome_additionalSources_0_amount"]', data.netWorth.additionalSources[0].amount)
+      .click('.va-growable-expanded .float-left');
+}
+
+function completeExpectedIncomeInfo(client, data) {
+  client
+      .fill('input[name="root_expectedIncome_salary"]', data.expectedIncome.salary)
+      .fill('input[name="root_expectedIncome_interest"]', data.expectedIncome.interest)
+      .click('.pensions-sources-add-btn')
+      .fill('input[name="root_expectedIncome_additionalSources_0_name"]', data.expectedIncome.additionalSources[0].name)
+      .fill('input[name="root_expectedIncome_additionalSources_0_amount"]', data.expectedIncome.additionalSources[0].amount)
+      .click('.va-growable-expanded .float-left');
+}
+
+function completeOtherExpensesInfo(client, data) {
+  client
+      .selectYesNo('root_view:hasOtherExpenses', data['view:hasOtherExpenses'])
+      .fill('input[name="root_otherExpenses_0_amount"]', data.otherExpenses[0].amount)
+      .fill('input[name="root_otherExpenses_0_purpose"]', data.otherExpenses[0].purpose)
+      .fillDate('root_otherExpenses_0_date', data.otherExpenses[0].date)
+      .fill('input[name="root_otherExpenses_0_paidTo"]', data.otherExpenses[0].paidTo);
+}
+
+function completeSpouseNetWorthInfo(client, data) {
+  client
+      .fill('input[name="root_spouseNetWorth_bank"]', data.spouseNetWorth.bank)
+      .fill('input[name="root_spouseNetWorth_interestBank"]', data.spouseNetWorth.interestBank)
+      .fill('input[name="root_spouseNetWorth_ira"]', data.spouseNetWorth.ira)
+      .fill('input[name="root_spouseNetWorth_stocks"]', data.spouseNetWorth.stocks)
+      .fill('input[name="root_spouseNetWorth_realProperty"]', data.spouseNetWorth.realProperty)
+      .click('.pensions-sources-add-btn')
+      .fill('input[name="root_spouseNetWorth_additionalSources_0_name"]', data.spouseNetWorth.additionalSources[0].name)
+      .fill('input[name="root_spouseNetWorth_additionalSources_0_amount"]', data.spouseNetWorth.additionalSources[0].amount)
+      .click('.va-growable-expanded .float-left');
+}
+
+function completeSpouseMonthlyIncomeInfo(client, data) {
```

Comment:
Chris made this suggestion the other day; you can match on the end of a name property:

`input[name$="_socialSecurity"]`

I think if you used that and passed in the specific object for each one (e.g. pass `data.spouseMonthlyIncome` instead of `data` to this function), you could remove some of the duplication.